File: src/main/java/me/leodev/trollingfreedomreborn/other/UntrollOnQuit.java
package me.leodev.trollingfreedomreborn.other;

import me.leodev.trollingfreedomreborn.commands.UnTroll;
import me.leodev.trollingfreedomreborn.main.Core;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

import java.io.IOException;

public class UntrollOnQuit implements Listener {

    @EventHandler
    public void onQuit(PlayerQuitEvent event) {
        // Check if the feature is enabled in config.yml
        if (Core.instance.getConfig().getBoolean("values.untroll-on-quit", true)) {
            Player player = event.getPlayer();

            try {
                // Initialize UnTroll and stop tasks/lists for this specific player
                UnTroll untroll = new UnTroll();
                // Passing null for sender because the server/plugin is triggering it
                untroll.StopTrolls(player, null);
                Bukkit.getConsoleSender().sendMessage("§b§lTFR §8| §7Auto-untrolled §f" + player.getName() + " §7due to disconnect.");
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/other/UntrollOnQuit.kt
package com.leomadrassi.trollingfreedomreborn.other

import com.leomadrassi.trollingfreedomreborn.commands.UnTroll
import com.leomadrassi.trollingfreedomreborn.main.Core
import org.bukkit.Bukkit
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

class UntrollOnQuit : Listener {

    @EventHandler
    fun onQuit(event: PlayerQuitEvent) {
        if (Core.instance.config.getBoolean("values.untroll-on-quit", true)) {
            val player = event.player
            try {
                UnTroll().StopTrolls(player, null)
                Bukkit.getConsoleSender().sendMessage("§b§lTFR §8| §7Auto-untrolled §f${player.name} §7due to disconnect.")
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }
}
